ibis/pyspark/tests/test_basic.py

```diff
@@ -0,0 +1,168 @@
+import pandas as pd
+import pandas.util.testing as tm
+import pytest
+
+import ibis
+
+pytestmark = pytest.mark.pyspark
+
+
+@pytest.fixture(scope='session')
+def client():
+    pytest.importorskip('pyspark')
+    from pyspark.sql import SparkSession
+    import pyspark.sql.functions as F
+
+    session = SparkSession.builder.getOrCreate()
+    client = ibis.pyspark.connect(session)
+    df = client._session.range(0, 10)
+    df = df.withColumn("str_col", F.lit('value'))
+    df.createTempView('table1')
+
+    df1 = client._session.createDataFrame([(True,), (False,)]).toDF('v')
+    df1.createTempView('table2')
+    return client
+
+
+def test_basic(client):
+    table = client.table('table1')
+    result = table.compile().toPandas()
+    expected = pd.DataFrame({'id': range(0, 10), 'str_col': 'value'})
+
+    tm.assert_frame_equal(result, expected)
+
+
+def test_projection(client):
+    table = client.table('table1')
+    result1 = table.mutate(v=table['id']).compile().toPandas()
+
+    expected1 = pd.DataFrame(
+        {
+            'id': range(0, 10),
+            'str_col': 'value',
+            'v': range(0, 10),
+        }
+    )
+
+    result2 = (
+        table
+        .mutate(v=table['id'])
+        .mutate(v2=table['id'])
+        .mutate(id=table['id'] * 2)
+        .compile().toPandas()
+    )
+
+    expected2 = pd.DataFrame(
+        {
+            'id': range(0, 20, 2),
+            'str_col': 'value',
+            'v': range(0, 10),
+            'v2': range(0, 10),
+        }
+    )
+
+    tm.assert_frame_equal(result1, expected1)
+    tm.assert_frame_equal(result2, expected2)
+
+
+def test_aggregation_col(client):
+    table = client.table('table1')
+    result = table['id'].count().execute()
+    assert result == table.compile().count()
+
+
+def test_aggregation(client):
+    import pyspark.sql.functions as F
+
+    table = client.table('table1')
+    result = table.aggregate(table['id'].max()).compile()
+    expected = table.compile().agg(F.max('id'))
+
+    tm.assert_frame_equal(result.toPandas(), expected.toPandas())
+
+
+def test_groupby(client):
+    import pyspark.sql.functions as F
+
+    table = client.table('table1')
+    result = table.groupby('id').aggregate(table['id'].max()).compile()
+    expected = table.compile().groupby('id').agg(F.max('id'))
+
+    tm.assert_frame_equal(result.toPandas(), expected.toPandas())
+
+
+@pytest.mark.xfail
```

Comment:
Can you add the `reason` argument here?

Reply:
And also the `raises` argument, so that we know exactly why this test fails

Reply:
Added